addons/hr_homeworking/models/hr_homeworking.py

```diff
@@ -0,0 +1,306 @@
+    parent_default_homeworking = fields.Many2one("hr.employee.location", help="removed records from default record", ondelete='cascade')
+    child_removed_ids = fields.One2many("hr.employee.location", 'parent_default_homeworking')
+    current = fields.Boolean(default=True, help="useful only if it's a default worklocation or old default")  # transform into compute
+    end = fields.Date(help="useful only if it's an old default worklocation to save the past")
+
+    _sql_constraints = [
+        ('uniq_current', 'unique(employee_id, weekday, current)', 'Only one current default work location per day per employee. can have some previous default work location'),
+        ('uniq_exceptional_per_day', 'unique(employee_id, weekly, start_date, current)', 'Only one default work location and one exceptional work location per day per employee.'),
+    ]
+    @api.depends('start_date', 'weekly')
+    def _compute_weekday(self):
+        for record in self:
+            if not record.weekly:
+                continue
+            if record.start_date:
+                record.weekday = record.start_date.weekday()
+                record.employee_id.sudo().with_context(no_loop=True).write({DAYS[record.weekday]: record.work_location_id})
+
+    @api.depends('start_date', 'end_date_create')
+    def _compute_today_next_date(self):
+        for record in self:
+            if record.start_date < fields.Date().today():
+                if record.end_date_create >= fields.Date().today():
+                    record.today_next_date = fields.Date().today()
+                else:
+                    if record.start_date.weekday() <= fields.Date().today().weekday() and fields.Date().today().weekday() <= record.end_date_create.weekday():
+                        record.today_next_date = fields.Date().today()
+                    else:
+                        today = fields.Date().today()
+                        offset = (record.start_date.weekday() - fields.Date.today().weekday() + 7) % 7
+                        record.today_next_date = today + timedelta(days=offset)
+            else:
+                record.today_next_date = record.start_date
+
+    @api.depends('start_date', 'end_date_create', 'weekly')
+    def _compute_day_week_string(self):
+        for record in self:
+            if record.weekly and record.end_date_create - record.start_date > timedelta(days=7):
+                raise ValidationError(_('A weekly location cannot have a duration of more than 7 days'))
+            if (record.start_date > record.end_date_create):
+                record.end_date_create = record.start_date
+            weekdays_name = ""
+            if record.start_date.weekday() > record.end_date_create.weekday():
+                for i in range(0, record.end_date_create.weekday()):
+                    weekdays_name = weekdays_name + ", " + DAYS_STRING[i]
```

Comment:
This code will be overriden with the next loop, to be removed probably?

Reply:
no, the 2 loops complete each other in case start date is a later day than the end date